LuceneRDD does not handle null/missed values

Caused by: scala.MatchError: null
  at org.zouzias.spark.lucenerdd.package$.typeToDocument(package.scala:65)
  at org.zouzias.spark.lucenerdd.package$$anonfun$sparkRowToDocument$1.apply(package.scala:123)
  at org.zouzias.spark.lucenerdd.package$$anonfun$sparkRowToDocument$1.apply(package.scala:121)
  at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
  at org.zouzias.spark.lucenerdd.package$.sparkRowToDocument(package.scala:121)
  at org.zouzias.spark.lucenerdd.LuceneRDD$$anonfun$apply$5.apply(LuceneRDD.scala:316)
  at org.zouzias.spark.lucenerdd.LuceneRDD$$anonfun$apply$5.apply(LuceneRDD.scala:316)
  at org.zouzias.spark.lucenerdd.partition.LuceneRDDPartition$$anonfun$3.apply(LuceneRDDPartition.scala:47)
  at org.zouzias.spark.lucenerdd.partition.LuceneRDDPartition$$anonfun$3.apply(LuceneRDDPartition.scala:45)
  at scala.collection.Iterator$class.foreach(Iterator.scala:742)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1194)
  at org.zouzias.spark.lucenerdd.partition.LuceneRDDPartition.<init>(LuceneRDDPartition.scala:45)
  at org.zouzias.spark.lucenerdd.partition.LuceneRDDPartition$.apply(LuceneRDDPartition.scala:136)
  at org.zouzias.spark.lucenerdd.LuceneRDD$$anonfun$7.apply(LuceneRDD.scala:289)

--- a/src/main/scala/org/zouzias/spark/lucenerdd/package.scala
+++ b/src/main/scala/org/zouzias/spark/lucenerdd/package.scala
@@ -16,8 +16,7 @@
  */
 package org.zouzias.spark
 
-import org.apache.lucene.document.{DoubleDocValuesField, _}
-import org.apache.lucene.facet.FacetField
+import org.apache.lucene.document._
 import org.apache.spark.sql.Row
 
 import scala.reflect.ClassTag
@@ -29,31 +28,31 @@ package object lucenerdd {
 
   implicit def intToDocument(v: Int): Document = {
     val doc = new Document
-    doc.add(new IntField(DefaultFieldName, v, Stored))
+    if (v != null) doc.add(new IntField(DefaultFieldName, v, Stored))
     doc
   }
 
   implicit def longToDocument(v: Long): Document = {
     val doc = new Document
-    doc.add(new LongField(DefaultFieldName, v, Stored))
+    if (v != null) doc.add(new LongField(DefaultFieldName, v, Stored))
     doc
   }
 
   implicit def doubleToDocument(v: Double): Document = {
     val doc = new Document
-    doc.add(new DoubleField(DefaultFieldName, v, Stored))
+    if (v != null)  doc.add(new DoubleField(DefaultFieldName, v, Stored))
     doc
   }
 
   implicit def floatToDocument(v: Float): Document = {
     val doc = new Document
-    doc.add(new FloatField(DefaultFieldName, v, Stored))
+    if (v != null) doc.add(new FloatField(DefaultFieldName, v, Stored))
     doc
   }
 
   implicit def stringToDocument(s: String): Document = {
     val doc = new Document
-    doc.add(new TextField(DefaultFieldName, s, Stored))
+    if (s != null) doc.add(new TextField(DefaultFieldName, s, Stored))
     doc
   }
 
@@ -63,16 +62,17 @@ package object lucenerdd {
 
   def typeToDocument[T: ClassTag](doc: Document, fieldName: String, s: T): Document = {
     s match {
-      case x: String =>
+      case x: String if x != null =>
         doc.add(new TextField(fieldName, x, Stored))
-      case x: Long =>
+      case x: Long if x != null =>
         doc.add(new LongField(fieldName, x, Stored))
-      case x: Int =>
+      case x: Int if x != null =>
         doc.add(new IntField(fieldName, x, Stored))
-      case x: Float =>
+      case x: Float if x != null =>
         doc.add(new FloatField(fieldName, x, Stored))
-      case x: Double =>
+      case x: Double if x != null =>
         doc.add(new DoubleField(fieldName, x, Stored))
+      case _ => Unit
     }
     doc
   }
@@ -86,7 +86,7 @@ package object lucenerdd {
   implicit def mapToDocument[T: ClassTag](map: Map[String, T]): Document = {
     val doc = new Document
     map.keys.foreach{ case key =>
-      typeToDocument(doc, key, map.get(key).get)
+      typeToDocument(doc, key, map(key))
     }
     doc
   }

--- a/src/main/scala/org/zouzias/spark/lucenerdd/package.scala
+++ b/src/main/scala/org/zouzias/spark/lucenerdd/package.scala
@@ -85,8 +85,8 @@ package object lucenerdd {
 
   implicit def mapToDocument[T: ClassTag](map: Map[String, T]): Document = {
     val doc = new Document
-    map.keys.foreach{ case key =>
-      typeToDocument(doc, key, map(key))
+    map.foreach{ case (key, value) =>
+      typeToDocument(doc, key, value)
     }
     doc
   }

--- a/src/test/scala/org/zouzias/spark/lucenerdd/LucenePrimitiveTypesSpec.scala
+++ b/src/test/scala/org/zouzias/spark/lucenerdd/LucenePrimitiveTypesSpec.scala
@@ -104,4 +104,12 @@ class LucenePrimitiveTypesSpec extends FlatSpec with Matchers
     luceneRDD.prefixQuery("b", "wor").nonEmpty should equal (true)
     luceneRDD.prefixQuery("a", "no").nonEmpty should equal (false)
   }
+
+  "LuceneRDD" should "work with RDD[String] and ignore null values" in {
+    val array = Array("aaa", null, "ccc", null, "eee")
+    val rdd = sc.parallelize(array)
+    luceneRDD = LuceneRDD(rdd)
+    luceneRDD.count should be (array.size)
+  }
+
 }

--- a/src/test/scala/org/zouzias/spark/lucenerdd/LuceneRDDCustomCaseClassImplicitsSpec.scala
+++ b/src/test/scala/org/zouzias/spark/lucenerdd/LuceneRDDCustomCaseClassImplicitsSpec.scala
@@ -42,6 +42,14 @@ class LuceneRDDCustomCaseClassImplicitsSpec extends FlatSpec
   val elem = Array("fear", "death", "water", "fire", "house")
     .zipWithIndex.map{ case (str, index) => Person(str, index, s"${str}@gmail.com")}
 
+  "LuceneRDD(case class).count" should "handle nulls properly" in {
+    val elemsWithNulls = Array("fear", "death", "water", "fire", "house")
+      .zipWithIndex.map{ case (str, index) => Person(str, index, null)}
+    val rdd = sc.parallelize(elemsWithNulls)
+    luceneRDD = LuceneRDD(rdd)
+    luceneRDD.count should equal (elemsWithNulls.size)
+  }
+
   "LuceneRDD(case class).count" should "return correct number of elements" in {
     val rdd = sc.parallelize(elem)
     luceneRDD = LuceneRDD(rdd)

